### Integration and optimization

Use AGAT (v0.6.0) (https://github.com/NBISweden/AGAT) to integrate the gene models generated by evidence-based 
prediction and ab initio prediction into a set of unique gene models (using gene_Model.sh scripts). These gene models were 
submitted to PASA (v2.4.1) to perform refined process (using pasa_1.sh, pasa_2.sh and pasa_3.sh scripts). The B73 full-length cDNA,
maize transcripts generated by Iso-seq, maize EST sequences in GeneBank and the transcripts fasta file generated by 
Mikado (v2.4.4) were used in the PASA analysis. The refined gene models were submitted to TEsorter (v1.3) to filter 
out the TE-related gene (using TE_filter.sh script). The TE-free gene models were submitted to MAKER-P (v3.01.03) to assign AED score. 
The gene models of which AED is less than 0.5 were kept (get_AED_0.5_res.sh). Then a series of analysis was used to
pick the high-confidence gene models, check and fix the potential mistake in gff3 file format (using post_AED_0.5_process.sh and 
get_final_gff3.sh scripts). The AGAT (v0.6.0) and the GFF3toolkit (v2.0.3) were used in this process.
