### Evidence-based prediction

#### Use STAR (v2.7.7a) to align the RNA-seq data (star_index.sh, star_align_rnd1.sh, star_align_rnd2.sh) to find the potential splice junction and generate bam file. Then the bam file was submitted to Trinity (v2.11.0), StringTie (v2.0.6), Strawberry (v1.1.2), Cufflinks (v2.2.1), Class2 (v2.1.7) to assembly transcripts (stringtTie.sh, strawberry.sh, cufflinks.sh, class2.sh, trinity.sh). And the GMAP (version 2017-11-15) was used to align the fasta file generated by Trinity (v2.11.0) to genome in order to get the gtf file (run_gmap.sh). The assembled transcripts encoded in gtf file were submitted Mikado (v2.4.4) to pick the optimal transcripts set (mikado_1.sh, mikado_2.sh). Besides, the high-confidence splice junction generated by Portcullis (v1.2.0), the ORFs predicted by TransDecoder (transDecoder.sh) and the homology results between assembled transcripts and SwissProt (viridiplantae) (xml_processing.sh, blastx.sh, BlastXMLmerge.py).
